contract_mandate/models/contract.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2017 Carlos Dauden - Tecnativa <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ContractContract(models.Model):
+    _inherit = 'contract.contract'
+
+    mandate_id = fields.Many2one(
+        comodel_name='account.banking.mandate',
+        ondelete='restrict',
+        string='Direct Debit Mandate',
+        help="If mandate required in payment method and not set mandate, "
+             "invoice takes the first valid mandate",
+        index=True,
+    )
+    mandate_required = fields.Boolean(
+        related='payment_mode_id.payment_method_id.mandate_required',
+        readonly=True)
```

Comment:
@ThomasBinsfeld @aheficent missing string And the string should be `Is Mandate Required`